types/Luau.luau

```diff
@@ -0,0 +1,116 @@
+--[=[
+	@interface CompileSettings
```

Comment:
Nitpick - we should probably do `Options` instead of `Settings` for these names to keep things consistent across the built-in libraries. Same goes for the function arg names below